src/code-components/RichTextArea/RichTextArea.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useMemo, useRef } from "react";
+import Quill, { Range } from "quill/core";
+import {
+  formatDefaultToolbarConfigs,
+  Toolbar,
+  ToolbarConfigs,
+} from "./formatDefaultToolbarConfigs";
+import dynamic from "next/dynamic";
+
+interface RichTextAreaProps {
+  htmlValue?: string;
+  toolbar: Toolbar;
+  customToolbar?: ToolbarConfigs;
+  onSelectionChange?: (range: Range | null, source: string) => void;
+  onChange?: (content: string, source: string) => void;
+  onBlur?: (range: Range | null, source: string) => void;
+  onFocus?: (range: Range | null, source: string) => void;
+  onKeyDown?: (event: KeyboardEvent) => void;
+  onKeyUp?: (event: KeyboardEvent) => void;
+  placeholder?: string;
+  readOnly: boolean;
+  wrapperClassName: string;
+  ariaLabel?: string;
+  ariaLabeledby?: string;
+  role?: string;
+}
+
+export function RichTextArea({
+  htmlValue,
+  toolbar,
+  customToolbar,
+  onSelectionChange,
+  onChange,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  placeholder,
+  readOnly,
+  wrapperClassName,
+  ariaLabel,
+  ariaLabeledby,
+  role,
+}: RichTextAreaProps) {
+  const Editor = useMemo(() => {
+    return dynamic(() => import("./Editor").then((module) => module.Editor), {
```

Comment:
uaa, a dynamic import... i'm worried it may cause us problems some time later.

it works? how exactly? instead of this dynamic import magic, perhaps we can just mount it always (no matter what environment)?

and/or use `useState` with `useEffect` to mount the editor after client render/hydration kicks in.

Reply:
tldr: get rid of dynamic import and just mount the editor always.

if quill can't be imported on the server for any reason, it's a problem of `Editor` component (not the rich text area), and we can get around that, by rendering a placeholder on initial render, and later rendering quill after client render comes (`useEffect(() => setIsMounted(true)); return isMounted ? quill : placeholder;`)